addons/mail/models/mail_template.py

```diff
@@ -585,13 +585,30 @@ def send_mail(self, res_id, force_send=False, raise_exception=False, email_value
 
         # Grant access to send_mail only if access to related document
         self.ensure_one()
-        self._send_check_access([res_id])
+        return self.send_mail_batch(
+            [res_id],
+            force_send=force_send,
+            raise_exception=raise_exception,
+            email_values=email_values,
+            email_layout_xmlid=email_layout_xmlid
+        )[0].id  # TDE CLEANME: return mail + api.returns ?
+
+    @api.returns('self', lambda value: value.ids)
+    def send_mail_batch(self, res_ids, force_send=False, raise_exception=False, email_values=None,
```

Comment:
it's a bit mhe to have a batch method and a non batch method imo, it's like having `create` and `create_batch` (the batched method can support both flow)

(especially if we return a recordset)

Reply:
I did not want to add an input/output converter (I know you did it, but I don't really like that). But for this "well known" method I can do it if you prefer, and remove it in master to be only batch enabled and return recordsets.

Reply:
ok to do it in master :)